single_kernel_mongo/managers/mongodb_operator.py

```diff
@@ -463,7 +565,9 @@ def update_config_and_restart(self) -> None:
                 "Invalid LDAP Query template, please update your config."
             )
 
-        if self.state.upgrade_in_progress:
+        if self.refresh is None:
+            logger.warning("Warning update_config_and_restart: Refresh could be in progress.")
```

Comment:
should we not return here? if refresh is none

Reply:
I think we want to defer because otherwise we would lose the event.
However, after reading the code, this line is useless as `self.refresh_in_progress` considers that not having the `self.refresh` object indicates a refresh could be in progress.
I'll remove this line.